.github/workflows/dependency-check.yml

```diff
@@ -0,0 +1,25 @@
+name: Dependency Check
+
+on: [push, pull_request]
```

Comment:
```suggestion
on:
  push:
    branches:
      - main
  pull_request:
    branches:
      - main
```

Reply:
without specifying branch for `push` event, the workflow will be triggered for any push into any branch. This will make the workflow to run twice when commits are pushed to PR branch.

Reply:
or you can simply remove `push` event here, since the PR cannot be merged without syncing with main branch and running this workflow again on main branch doesn't make sense.